lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -453,26 +474,27 @@ def __get_k8s_ingress_spec(self, ajs):
                 "annotations": {"app.galaxyproject.org/tool_id": ajs.job_wrapper.tool.id},
             },
             "spec": {
-                "rules": [
-                    {
-                        "host": ep["domain"],
-                        "http": {
-                            "paths": [
-                                {
-                                    "backend": {
-                                        "serviceName": self.__get_k8s_job_name(
-                                            self.__produce_k8s_job_prefix(), ajs.job_wrapper
-                                        ),
-                                        "servicePort": int(ep["tool_port"]),
-                                    },
-                                    "path": ep.get("entry_path", "/"),
-                                    "pathType": "Prefix",
-                                }
-                            ]
-                        },
-                    }
-                    for ep in entry_points
-                ]
+                "ingressClassName": "nginx",
```

Comment:
```suggestion
```

I think it's better to let the cluster default ingress class apply?

Reply:
Yes, I might have forced this when it wasn't working and I was trying many different things. I will check if removing it keeps the functionality (I expect it should).

Reply:
...mmm... no, unrelated. Either one of the commits in the rebase or something else.

Reply:
...found it now.

Reply:
Could you please remove the commit suggestion that deletes the ingress class above so that no one presses it by mistake :-) ? Thanks!

Reply:
Have you defined a default ingressclass on your cluster? It's probably breaking because one isn't defined?

Reply:
Most likely not, but probably not ideal to expect that every cluster has one defined. I would go for config with a reasonable default instead of removal.

Reply:
I don't think that cluster misconfiguration is something that we need to actively handle in the chart. It doesn't look like we can reasonably specify a default, because the default ingress class applies only if no ingress class is defined? Are you proposing adding a parameter to explicitly specify the ingress class if needed, and if not, just uses the default ingress class? That's ok I guess, although it could also be somewhat redundant if the cluster admin can simply set the default ingress class instead. We should add a note in the readme that a default ingress class should be defined.

Reply:
Well, defining or not a default ingress class is not necessarily misconfiguration. It really depends on the version of k8s you are using, the helm chart that was used to deploy the ingress class, etc. I remember reading various issues around people having or not having their default ingress class set, or sometimes defining them and that still not working on some versions. Even the helm chart actively specifies ingressClassName on some templates and doesn't leave that to the cluster having a default set or not. I would be defensive here.